bonsai-core/src/main/scala/com/stripe/bonsai/FullBinaryTree.scala

```diff
@@ -6,36 +6,123 @@ import scala.collection.immutable.Queue
 
 import com.stripe.bonsai.layout.Vec
 
+/**
+ * This type represents a binary tree, where each branch is guaranteed
+ * to have two children (i.e. it is "full").
+ *
+ * =Members=
+ *
+ * {{{
+ *  - bitset: encodes the tree structure using a concept of internal
+ *    and external nodes, e.g. a (potentially unbalanced) heap.  it's
+ *    sort of complicated. basically, for non-balanced trees, we need
+ *    to add "spacing" so that the heap addressing invariants are
+ *    preserved.
+ *
+ *  - isLeaf: encodes whether an (internal) node is a leaf or branch
+ *    node.
+ *
+ *  - branchLabels: a dense array of branch label values to index
+ *    into. we use isLeaf to convert 'bitset indices' into
+ *    'branchLabels' indices.
+ *
+ *  - leafLabels: a dense array of leaf label values to index into. we
+ *    use isLeaf to convert 'bitset indices' into 'leafLabels' indices.
+ * }}}
+ *
+ * =Examples=
+ *
+ * ==Example 1==
+ *
+ * {{{
+ *       a
+ *      / \
+ *     b   c
+ *    / \ / \
+ *   d  e f  g
+ *
+ * (each leaf has two fake "external nodes" which are 0's in bitset)
+ *
+ * bitset: 111111100000000
+ * isLeaf: 0001111
+ * branchLabels: a,b,c
+ * leafLabels: d,e,f,g
+ * }}}
+ *
+ * ==Example 2==
+ *
+ * {{{
+ *       a
+ *      / \
+ *     b   c
+ *        / \
+ *        d  e
+ *
+ * (each leaf has two fake "external nodes" which are 0's in bitset)
+ *
+ * bitset: 11100110000
+ * isLeaf: 01011
+ * branchLabels: a,c
+ * leafLabels: b,d,e
+ * }}}
+ *
+ */
 class FullBinaryTree[A, B](
```

Comment:
Want to open an issue about variance? I'd be happy to address it in a future PR.